akka-http/src/main/scala/akka/http/scaladsl/server/RouteResult.scala

```diff
@@ -72,3 +72,17 @@ object RouteResult {
     Route.handlerFlow(route)(routingSettings, parserSettings, materializer, routingLog, ec, rejectionHandler, exceptionHandler)
   }
 }
+
+sealed abstract class LowerPriorityRouteResultImplicits {
+  /**
+   * Turns a `Route` into a server flow.
+   *
+   * This implicit conversion is defined here because `Route` is an alias for
+   * `RequestContext => Future[RouteResult]`, and the fact that `RouteResult`
+   * is in that type means this implicit conversion come into scope whereever
+   * a `Route` is given but a `Flow` is expected.
+   */
+  implicit def routeToFlowViaMaterializer(route: Route)(implicit materializer: Materializer): Flow[HttpRequest, HttpResponse, NotUsed] =
```

Comment:
One thing I wonder about, is if we should deprecate it right away, so we can remove it again in the next version?